### DISCUSSION TITLE
The purpose of the plugin's behavior in general

### DISCUSSION THREAD

**Discussion by Marwa-Da** [molstar/molstar#904] [Q&A] [answered]
Hi all,

I'm just wondering what do the behaviors do! You can take the accessibleSurfaceArea as an example. How to use it? What does the "eye" and the "auto attach" options do?

Any brief explanation will help and is much appreciated.
![image](https://github.com/molstar/molstar/assets/11466297/7d5d47af-d6af-48d3-82ee-2fa16df48442)

regards,

**Comment by dsehnal** [comment, marked as answer]
In general, the plugin behaviors modify how the plugin "behaves" - e.g. what happens when user clicks on a residue, what properties are available to compute/visualize for a given structure, etc. In this case, this behavior adds the option to compute and visualize the ASA.

Auto-attach means that that each time a new `Structure` node is created in the state tree, the ASA property is available for the user to compute and visualize. If auto-attach was false, the property would have to be added manually. Show tooltip means whether to display the ASA information in the hover tooltip in bottom right.

-----------

For behaviors, the eye icons don't do anything (unless a part of the behavior is the ability to react to changing of the visibility state, which can be coded). They only work to toggle visibility in the "main" state tree.
